Pat_SO3/Test_Pat_SO3.cpp

```diff
@@ -0,0 +1,66 @@
+#include "Pat_SO3.h"
+#include <iostream>
+#include <cassert>
+
+//Test if SO3
+void test_is_SO3() {
```

Comment:
I would add some negative test cases. For example, currently the matrix 
`[1 1 0;
 0 0 0;
 0 0 0]` will pass this test, which is definitely not an SO(3) element.